### PR TITLE
Add support for ALB auth

### PR DESCRIPTION
**Pull request by dynamike**
This PR modifies the atlantis terraform module to support authentication on the ALB using AWS Cognito. It adds two routing rules. 

1. If the source IP matches Github's static IP addresses, forward to the atlantis service unauthenticated. This will allow webhooks to continue to function.
2. For all paths and IPs, check if the user is authenticated against AWS Cognito, which is GSuite SAML in our case. Once the request is authenticated, forward it to the atlantis service.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,58 +126,71 @@ If all provided subnets are public (no NAT gateway) then `ecs_service_assign_pub
 * [GitLab repository webhook for Atlantis](https://github.com/terraform-aws-modules/terraform-aws-atlantis/tree/master/examples/gitlab-repository-webhook)
 
 <!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+## Providers
+
+| Name | Version |
```
Comment (esacteksab) [inline README.md]: I don't know if this is auto-generated, but maybe at least capture the values that exist today as of the creation of this PR?

File: main.tf [hunk 2]
```diff
@@ -88,6 +88,8 @@ locals {
   )
 }
 
+data "github_ip_ranges" "current" {}
```
Comment (esacteksab) [inline main.tf]: Woah! TIL!

File: README.md (hunk 1 above)
Comment (dynamike) [reply README.md]: those are generated from terraform-docs and specify the provider versions are required when using the module. I'm not sure what the required versions are

File: variables.tf [hunk 3]
```diff
@@ -78,6 +78,31 @@ variable "alb_logging_enabled" {
...
+
+variable "cognito_user_pool_client_id" {
+  description = "The ID of the Cognito user pool client"
+  type        = string
+  default     = ""
+}
+
+variable "cognito_user_pool_domain" {
+  description = "The domain prefix or fully-qualified domain name of the Cognito user pool"
+  type        = "string"
```
Comment (esacteksab) [inline variables.tf]: You have `string` above, but you have `"string"` here -- consistency pls!

Comment (esacteksab) [inline variables.tf]:
```suggestion
  type        = string
```

Comment (dynamike) [reply variables.tf]: doh

**Pull request merged**